data_structures/queue/priority_queue.py

```diff
@@ -0,0 +1,273 @@
+"""
+Pure Python implementation of Priority Queue using lists
+"""
+
+
+class Error(Exception):
+    """Base class for exceptions in this module."""
+
+    pass
+
+
+class OverFlow(Error):
+    def __init__(self, msg):
+        self.msg = msg
+
+
+class UnderFlow(Error):
+    def __init__(self, msg):
+        self.msg = msg
+
+
+class FixedPriorityQueue:
+    """
+    In a Priority Queue the elements are entred as an when the come
+    But while removing or deleting an element the highest priority element is deleted
+    in FIFO fashion.
+    Here the lowest integer has the highest priority.
+    Example:
+    priority(0) > priority(5)
+    priority(16) > priority(32)
+    Here as an example I have taken only 3 priorities viz. 0, 1, 2
+    0 Being the highest priority and 2 being the lowest
+    You can change the priorities as per your need
+
+    Examples
+    >>> FPQ = FixedPriorityQueue()
+    >>> FPQ.enqueue(0, 10)
+    >>> FPQ.enqueue(1, 70)
+    >>> FPQ.enqueue(0, 100)
+    >>> FPQ.enqueue(2, 1)
+    >>> FPQ.enqueue(2, 5)
+    >>> FPQ.enqueue(1, 7)
+    >>> FPQ.enqueue(2, 4)
+    >>> FPQ.enqueue(1, 64)
+    >>> FPQ.enqueue(0, 128)
+    >>> FPQ.print_queue()
```

Comment:
According to "Introduction to Algorithms" by Cormen, Leiserson, Rivest, and Stein, a priority queue is defined as follows:

> A priority queue is a data structure for maintaining a set `S` of elements each with an associated value called a `key`. A _max-priority queue_ supports the following operations:
>
> `Insert(S, x)`: inserts the element `x` into the set `S`, wich is equivalent to the operation `S = S ∪ {x}`.
> `Maximum(S)`: returns the element of `S` with the largest key.
> `Extract-Max(S)`: removes and returns the element of S with the largest key.
> `Increase-Key(S, x, k)`: increases the value of element `x`'s key to the new value `k`, which is assumed to be at least as large as `x`'s current key value.
> ...
> Alternatively, a _min-priority queue_ supports the operations `Insert`, `Minimum`, `Extract-Min`, and `Decrease-Key`. 

Priority queues often times use a heap as the underlying implementation. Python itself supports this with the built in [heapq module](https://docs.python.org/3/library/heapq.html). This priority queue list of lists is a non-standard implementation. My opinion is that it should not be included as an example of a priority queue in a repo about demonstrating algorithms and data structures.

Reply:
Oh, OK
I'll learn about heaps and their implementation in designing queues and then try again